python/src/spark_rapids_ml/umap.py

```diff
@@ -1283,15 +1354,29 @@ def _construct_umap() -> CumlT:
             internal_model = CumlUMAP(**cuml_alg_params)
             internal_model.embedding_ = cp.array(embedding).data
             internal_model._raw_data = raw_data_cuml
+            internal_model.sparse_fit = self.sparse_fit
 
             return internal_model
 
         def _transform_internal(
             umap: CumlT,
-            df: Union[pd.DataFrame, np.ndarray],
+            df: Union[pd.DataFrame, np.ndarray, scipy.sparse._csr.csr_matrix],
         ) -> pd.Series:
+            from cuml.common.sparse_utils import is_sparse
+
             embedding = umap.transform(df)
 
+            if is_sparse(df):
+                # scipy csr matrix to dense numpy array
+                df = df.toarray()  # type: ignore
```

Comment:
TODO: there's definitely a better way to return the inputs rather than reconvert to dense array. Either SparseVectors or COO format would probably be preferable if possible.